src/functions-and-arrays.js

```diff
@@ -71,3 +253,37 @@ const matrix = [
   [20, 73, 35, 29, 78, 31, 90, 1, 74, 31, 49, 71, 48, 86, 81, 16, 23, 57, 5, 54],
   [1, 70, 54, 71, 83, 51, 54, 69, 16, 92, 33, 48, 61, 43, 52, 1, 89, 19, 67, 48]
 ];
+
+
+function greatestProduct(arr) {
```

Comment:
You are close, but one test case is failing: 
"should return 16 when all the numbers of the arrays are 2"
Can you find out why? If not, we can try to take a look at it together.